### PR TITLE
Add `BuddyRemoveAll` DBus method to clear friend-list

### PR DESCRIPTION
**Pull request by Kavan72**
@Levak i added removeall method in DBus API. just check all things

### CODE REVIEW

File: src/dbus/methods/buddy_remove_all.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+#include <wb_dbus_methods.h>
+
+#include <wb_session.h>
+#include <wb_list.h>
+#include <wb_friend.h>
+#include <wb_cmd.h>
+
+
+static void flist_to_array(struct friend *f, GVariantBuilder *builder)
```
Comment (Levak) [inline src/dbus/methods/buddy_remove_all.c]:
```suggestion
static void remove_friend(struct friend *f, GVariantBuilder *builder)
```

File: src/dbus/methods/buddy_remove_all.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+/*
+** DBus method call: "BuddyRemoveAll"
+*/
+gboolean on_handle_buddy_remove_all(Warfacebot *object,
+                           GDBusMethodInvocation *invocation)
+{
+    struct list *fl = session.profile.friends;
+
+    list_foreach(fl, (f_list_callback) flist_to_array, NULL);
```
Comment (Levak) [inline src/dbus/methods/buddy_remove_all.c]:
```suggestion
    list_foreach(fl, (f_list_callback) remove_friend, NULL);
```

File: src/dbus/methods/buddy_remove_all.c [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    warfacebot_complete_buddy_remove_all(
+        object,
+        invocation);
+
+    return TRUE;
+}
```
Comment (Levak) [inline src/dbus/methods/buddy_remove_all.c]:
```suggestion
}

```

**Comment by Kavan72** [comment]
done

**Pull request merged**